### DISCUSSION TITLE
Added typesafety for multiple tags

### DISCUSSION THREAD

**Discussion by JaapWijnen** [pointfreeco/swift-html#80] [Ideas]
Hey! Really like the library. I was playing around with it and saw that it was possible to do something like 

```
.document(
  .html(
    .body(
      .html()
    )
  )
)
```

And tags like `.div` and `.script` can be used anywhere in .document.
`.script` can also be used anywhere within `.html` which are all examples of invalid html. 
I was playing around a bit and it would be quite easy to improve on this behavior by
- introducing an extension on `ChildOf` for doctype. 
- remove general extensions from `ChildOf` for `script` tags (and some others) and move these instead to specific extensions of `ChildOf` (introducing a little bit more code) 
- move all tags that are only used inside a body tag to an extension of `ChildOf where Element == Tag.Body` All these static functions would also return a ChildOf<Tag.Body> instead of the previous `Node` type. Then we can use them nested as well! 

I've got this working locally and works quite nice! :) Let me know if you're interested in a PR for this behavior!

**Comment by stephencelis** [comment]
I'm going to convert this to a discussion. I think we're always open to more safety. Just have to be mindful of things that affect compile times :)
